city_scrapers/spiders/il_port_district.py

```diff
@@ -0,0 +1,193 @@
+
+                meeting_time = "9:00am"
+
+                start = self._parse_start(date, year, meeting_time)
+
+                title = meeting_types[i] if not self.special_meeting\
+                    else self.special_meeting + meeting_types[i]
+
+                classification = self._parse_classification(i, meeting_types[i])
+
+                agendas_links = self._parse_agendas_links(classification, start)
+
+                minutes_links = None
+
+                if classification == "Board":
+                    minutes_links = self._parse_minutes_links(start)
+                links = agendas_links if not minutes_links else\
+                    agendas_links + minutes_links
+
+                meeting = Meeting(
+                    title=title,
+                    description="",
+                    classification=classification,
+                    start=start,
+                    end=None,
+                    all_day=False,
+                    time_notes="",
+                    location=self.location,
+                    links=links,
+                    source=response.url
+                )
+
+                meeting["status"] = self._get_status(meeting)
+                meeting["id"] = self._get_id(meeting)
+
+                yield meeting
+
+    def parse_agendas(self, response):
+        file_names = response.xpath("//tr/td[@class='views-field views-field-title']/text()")\
+            .extract()
+        file_names = [x.strip("\n ") for x in file_names]
+        file_links = response.xpath("//tr/td/a[@class='file-download']/@href").extract()
+
+        date_pattern = r"(?<=Agenda)(.*?)(?=.pdf)"
+        agenda_dates = [re.findall(date_pattern, x)[0].replace('%20', ' ') for x in file_links]
+
+        self.files_list = []
+
+        for link, name, date in list(zip(file_links, file_names, agenda_dates)):
+            self.files_list.append({'title': name + date, 'href': link})
+
+    def parse_minutes(self, response):
+        rows = response.xpath("//tr")
+        self.minutes_dict = {}
+        for row in rows:
+            file_name = row.xpath(".//td[@class='views-field views-field-title']/text()")\
+                .extract_first()
+            if not file_name:
+                continue
+
+            file_name = file_name.strip("\n ")
+            file_name_dt = re.findall(r"(?:.*?)(?:\d{4})", file_name)[0]
+            file_name_dt = datetime.datetime.strptime(file_name_dt, "%B %d, %Y")
+
+            file_link = row.xpath(".//td[@class='views-field views-field-field-file']/a/@href")\
+                .extract_first()
+
+            self.minutes_dict.setdefault(file_name_dt.date(), file_link)
+
+    def _parse_classification(self, i, meeting_types):
+        if "board" in meeting_types.lower():
+            return BOARD
+        elif "committee" in meeting_types.lower():
+            return COMMITTEE
+        else:
+            return NOT_CLASSIFIED
+
+    def _parse_start(self, date, year, meeting_time):
+
```

Comment:
`self.changed_meeting_time` and `self.special_meeting` are only used in this function, so these and any other variables that don't need to be used across different requests should be passed as arguments to the function or handled separately.

`special_meeting` could be handled in a `_parse_title` function that checks the date string, because right now logic for parsing the titles and times is combined in this method and it makes it harder to read